### PR TITLE
InsightVMCloud: data_connector,parsers,datasamples

### PR DESCRIPTION
**Pull request by averbn**
## Proposed Changes

  - new connector

### CODE REVIEW

**Review by v-maudan — CHANGES_REQUESTED** [review]

File: .script/tests/KqlvalidationsTests/CustomTables/NexposeInsightVMCloud_assets_CL [hunk 1]
```diff
@@ -0,0 +1,121 @@
+{
```
[the remark this answers is not in the document]
Comment (averbn) [reply .script/tests/KqlvalidationsTests/CustomTables/NexposeInsightVMCloud_assets_CL:1]: fixed

File: .script/tests/KqlvalidationsTests/CustomTables/NexposeInsightVMCloud_vulnerabilities_CL [hunk 2]
```diff
@@ -0,0 +1,197 @@
+{
```
[the remark this answers is not in the document]
Comment (averbn) [reply .script/tests/KqlvalidationsTests/CustomTables/NexposeInsightVMCloud_vulnerabilities_CL:1]: fixed

File: Solutions/Rapid7InsightVM/Parsers/InsightVMAssets [hunk 3]
```diff
@@ -0,0 +1,43 @@
+// Usage Instruction :
```
[the remark this answers is not in the document]
Comment (averbn) [reply Solutions/Rapid7InsightVM/Parsers/InsightVMAssets:1]: fixed

File: Solutions/Rapid7InsightVM/Parsers/InsightVMVulnerabilities [hunk 4]
```diff
@@ -0,0 +1,62 @@
+// Usage Instruction :
```
[the remark this answers is not in the document]
Comment (averbn) [reply Solutions/Rapid7InsightVM/Parsers/InsightVMVulnerabilities:1]: fixed

File: Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json [hunk 5]
```diff
@@ -0,0 +1,143 @@
+{
+	"id":"InsightVMCloudAPI",
+	"title":"Rapid7 Insight Platform Vulnerability Management Reports",
+	"publisher":"Rapid7",
+	"descriptionMarkdown":"The [Rapid7 Insight VM](https://www.rapid7.com/products/insightvm/) Report data connector provides the capability to ingest Scan reports and vulnerability data into Azure Sentinel through the REST API from the  Rapid7 Insight platform (Managed in the cloud). Refer to [API documentation](https://docs.rapid7.com/insight/api-overview/) for more information. The connector provides ability to get events which helps to examine potential security risks, analyze your team's use of collaboration, diagnose configuration problems and more",
```
[the remark this answers is not in the document]
Comment (averbn) [reply Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json:5]: fixed

File: Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
...ions** area, choose the **Deploy to function app** button.\nIf you aren't already signed in, choose the Azure icon in the Activity bar, then in the **Azure: Functions** area, choose **Sign in to Azure**\nIf you're already signed in, go to the next step.\n5. Provide the following information at the prompts:\n\n\ta. **Select folder:** Choose a folder from your workspace or browse to one that contains your function app.\n\n\tb. **Select Subscription:** Choose the subscription to use.\n\n\tc. Select **Create new Function App in Azure** (Don't choose the Advanced option)\n\n\td. **Enter a globally unique name for the function app:** Type a name that is valid in a URL path. The name you type is validated to make sure that it's unique in Azure Functions. (e.g. InsightVMXXXXX).\n\n\te. **Select a runtime:** Choose Python 3.8.\n\n\tf. Select a location for new resources. For better performance and lower costs choose the same [region](https://azure.microsoft.com/regions/) where Azure Sentinel is located.\n\n6. Deployment will begin. A notification is displayed after your function app is created and the deployment package is applied.\n7. Go to Azure Portal for the Function App configuration."
```
[the remark this answers is not in the document]
Comment (averbn) [reply Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json:136]: fixed

File: Solutions/Rapid7InsightVM/Data Connectors/azuredeploy_Connector_InsightVMCloudAPI_AzureFunction.json [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+                        "APPLICATIONINSIGHTS_CONNECTION_STRING": "[reference(resourceId('microsoft.insights/components', variables('FunctionName')), '2015-05-01').ConnectionString]",
+                        "AzureWebJobsStorage": "[concat('DefaultEndpointsProtocol=https;AccountName=', toLower(variables('FunctionName')),';AccountKey=',listKeys(resourceId('Microsoft.Storage/storageAccounts', toLower(variables('FunctionName'))), '2019-06-01').keys[0].value, ';EndpointSuffix=',toLower(variables('StorageSuffix')))]",
+                        "WorkspaceID": "[parameters('WorkspaceID')]",
+                        "WorkspaceKey": "[parameters('WorkspaceKey')]",
+                        "InsightVMAPIKey": "[parameters('InsightVMAPIKey')]",
+                        "InsightVMCloudRegion": "[parameters('InsightVMCloudRegion')]",
+                        "logAnalyticsUri": "[variables('LogAnaltyicsUri')]",
+                        "WEBSITE_RUN_FROM_PACKAGE": "https://github.com/averbn/azure_sentinel_data_connectors/raw/main/insight-vm-cloud-azure-sentinel-data-connector/InsightVMCloudAPISentinelConn.zip"
```
[the remark this answers is not in the document]
Comment (averbn) [reply Solutions/Rapid7InsightVM/Data Connectors/azuredeploy_Connector_InsightVMCloudAPI_AzureFunction.json:169]: fixed

File: Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+			 }
+		  ]
+	   },
+	   {
+		  "title":"Option 1 - Azure Resource Manager (ARM) Template",
+		  "description":"Use this method for automated deployment of the Rapid7 Insight Vulnerability Management Report data connector using an ARM Tempate.\n\n1. Click the **Deploy to Azure** button below. \n\n\t[![Deploy To Azure](https://aka.ms/deploytoazurebutton)](https://aka.ms/sentinelinsightvmcloudazuredeploy)\n2. Select the preferred **Subscription**, **Resource Group** and **Location**. \n> **NOTE:** Within the same resource group, you can't mix Windows and Linux apps in the same region. Select existing resource group without Windows apps in it or create new resource group.\n3. Enter the **InsightVMAPIKey**, choose **InsightVMCloudRegion** and deploy. \n4. Mark the checkbox labeled **I agree to the terms and conditions stated above**. \n5. Click **Purchase** to deploy."
+	   },
+	   {
+		  "title":"Option 2 - Manual Deployment of Azure Functions",
+		  "description":"Use the following step-by-step instructions to deploy the Rapid7 Insight Vulnerability Management Repor data connector manually with Azure Functions (Deployment via Visual Studio Code)."
```
[the remark this answers is not in the document]
Comment (averbn) [reply Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json:132]: fixed

File: Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+	   },
+	   {
+		  "title":"",
+		  "description":"**2. Configure the Function App**\n\n1. In the Function App, select the Function App Name and select **Configuration**.\n2. In the **Application settings** tab, select ** New application setting**.\n3. Add each of the following application settings individually, with their respective string values (case-sensitive): \n\t\tInsightVMAPIKey\n\t\tInsightVMCloudRegion\n\t\tWorkspaceID\n\t\tWorkspaceKey\n3. Once all application settings have been entered, click **Save**."
```
[the remark this answers is not in the document]
Comment (averbn) [reply Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json:140]: fixed

File: Solutions/Rapid7InsightVM/Parsers/InsightVMAssets [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+                    "DvcOsSysName", os_system_name_s,
+                    "DvcOsType", os_type_s,
+                    "DvcOsVendor", os_vendor_s,
+                    "DvcModelNumber", os_version_s,
+                    "RiskScore", risk_score_d,
+                    "SevereVulnerabilitiesCount", severe_vulnerabilities_d,
+                    "TotalVulnerabilitiesCount", total_vulnerabilities_d,
+                    "Uid", unique_identifiers_s,
+                    "VulnerabilitiesSolutions", same_s,
+                    "DvcMacAddr", mac_s
```
[the remark this answers is not in the document]
Comment (averbn) [reply Solutions/Rapid7InsightVM/Parsers/InsightVMAssets:34]: fixed

File: Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json (hunk 9 above)
[the remark this answers is not in the document]
Comment (v-maudan) [reply Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json]: @averbn , Can you please update instruction for logAnalyticsUri (optional). Thanks!

File: Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (v-maudan) [reply Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json:136]: @averbn , It seems short url link for Azure Function App not yet updated.

**Comment by v-maudan** [comment]
@averbn , Can you please fix all PR review comments. Thanks!

File: Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json (hunk 9 above)
[the remark this answers is not in the document]
Comment (averbn) [reply Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json]: done

File: Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json [hunk 11]
```diff
@@ -0,0 +1,143 @@
+{
+	"id":"InsightVMCloudAPI",
+	"title":"Rapid7 Insight Platform Vulnerability Management Reports",
+	"publisher":"Rapid7",
+	"descriptionMarkdown":"The [Rapid7 Insight VM](https://www.rapid7.com/products/insightvm/) Report data connector provides the capability to ingest Scan reports and vulnerability data into Azure Sentinel through the REST API from the  Rapid7 Insight platform (Managed in the cloud). Refer to [API documentation](https://docs.rapid7.com/insight/api-overview/) for more information. The connector provides ability to get events which helps to examine potential security risks, analyze your team's use of collaboration, diagnose configuration problems and more",
+	"additionalRequirementBanner":"These queries and workbooks are dependent on a parser based on Kusto to work as expected. ​Follow the steps to use this Kusto functions aliases [InsightVMAssets](https://aka.ms/) and [InsightVMVulnerabilities](https://aka.ms/)  in queries and workbooks",
```
[the remark this answers is not in the document]
Comment (v-maudan) [reply Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json:6]: @averbn ,Please update additionalRequirementBanner value. Thanks!

File: Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+	   {
+		  "title":"",
+		  "description":">**NOTE:** This connector uses Azure Functions to connect to the Insight VM API to pull its logs into Azure Sentinel. This might result in additional data ingestion costs. Check the [Azure Functions pricing page](https://azure.microsoft.com/pricing/details/functions/) for details."
+	   },
+	   {
+		  "title":"",
+		  "description":">**(Optional Step)** Securely store workspace and API authorization key(s) or token(s) in Azure Key Vault. Azure Key Vault provides a secure mechanism to store and retrieve key values. [Follow these instructions](https://docs.microsoft.com/azure/app-service/app-service-key-vault-references) to use Azure Key Vault with an Azure Function App."
+	   },
+	   {
+		  "description":">**NOTE:** This data connector depends on a parser based on a Kusto Function to work as expected. Follow the steps to create the Kusto functions aliases: [InsightVMAssets](https://aka.ms/) and [InsightVMVulnerabilities](https://aka.ms/)"
```
[the remark this answers is not in the document]
Comment (v-maudan) [reply Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json:96]: @averbn, can you please update parser description? Thanks!

File: Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (averbn) [reply Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json]: Already fixed, pls check the "InsightVM: connector json fix"  commit.

File: Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json [hunk 13]
```diff
@@ -0,0 +1,143 @@
...
+				   ],
+				   "label":"Primary Key"
+				},
+				"type":"CopyableLabel"
+			 }
+		  ]
+	   },
+	   {
+		  "title":"Option 1 - Azure Resource Manager (ARM) Template",
+		  "description":"Use this method for automated deployment of the Rapid7 Insight Vulnerability Management Report data connector using an ARM Tempate.\n\n1. Click the **Deploy to Azure** button below. \n\n\t[![Deploy To Azure](https://aka.ms/deploytoazurebutton)](https://aka.ms/sentinelinsightvmcloudazuredeploy)\n2. Select the preferred **Subscription**, **Resource Group** and **Location**. \n> **NOTE:** Within the same resource group, you can't mix Windows and Linux apps in the same region. Select existing resource group without Windows apps in it or create new resource group.\n3. Enter the **InsightVMAPIKey**, choose **InsightVMCloudRegion** and deploy. \n4. Mark the checkbox labeled **I agree to the terms and conditions stated above**. \n5. Click **Purchase** to deploy."
```
[the remark this answers is not in the document]
Comment (v-maudan) [reply Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json:128]: @averbn , Replace value https://aka.ms/sentinelinsightvmcloudazuredeploy with  https://aka.ms/sentinel-InsightVMCloudAPI-azuredeploy short url link for azure deploy .

File: Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (v-maudan) [reply Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json]: Update short url as per given format : https://aka.ms/sentinel-InsightVMCloudAPI-functionapp.

[the remark this answers is not in the document]
Comment (averbn) [reply Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json]: done

File: Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json (hunk 12 above)
[the remark this answers is not in the document]
Comment (v-maudan) [reply Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json:96]: @averbn , Can you please update on this .

File: Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json (hunk 13 above)
[the remark this answers is not in the document]
Comment (v-maudan) [reply Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json:128]: @averbn , Can you please update on this .

[the remark this answers is not in the document]
Comment (averbn) [reply Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json:128]: done

File: Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json (hunk 12 above)
[the remark this answers is not in the document]
Comment (v-maudan) [reply Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json:96]: @averbn , I am not able to see parser instruction description updated, can you please update as mentioned.

[the remark this answers is not in the document]
Comment (averbn) [reply Solutions/Rapid7InsightVM/Data Connectors/InsightVMCloud_API_FunctionApp.json:96]: done

**Review by v-maudan — APPROVED** [review]

**Review by v-jayakal — APPROVED** [review]

**Pull request merged**